### PR TITLE
Flex trip pattern key update

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR updates the trip pattern key equality check so that flex start/end pickup and drop off windows are also considered. This then creates additional patterns based on these values. Prior to this update trips were being grouped under the same, first, pattern which meant trips would have the wrong start/end pickup and drop off windows.

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]
Works well for me. This can help users a lot to differentiate patterns.

File: src/main/java/com/conveyal/gtfs/TripPatternKey.java [hunk 1]
```diff
@@ -99,6 +102,8 @@ public int hashCode() {
         result = 31 * result + (stops != null ? stops.hashCode() : 0);
         result = 31 * result + (pickupTypes != null ? pickupTypes.hashCode() : 0);
         result = 31 * result + (dropoffTypes != null ? dropoffTypes.hashCode() : 0);
+        result = 31 * result + (start_pickup_dropoff_window != null ? start_pickup_dropoff_window.hashCode() : 0);
```
Comment (miles-grant-ibigroup) [inline src/main/java/com/conveyal/gtfs/TripPatternKey.java]: Does the 31 have to change now that we have added variables? Isn't there now a clash potential

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks much cleaner! Just one quick question about the hashing

File: src/main/java/com/conveyal/gtfs/TripPatternKey.java (hunk 1 above)
Comment (br648) [reply src/main/java/com/conveyal/gtfs/TripPatternKey.java]: Possibility. I've updated the method to use the newer approach to this:

`Objects.hash(routeId, stops, pickupTypes, dropoffTypes, start_pickup_dropoff_window, end_pickup_dropoff_window);`

**Pull request merged**